### PR TITLE
chore(postgresql_metrics source): remove expect for postgreqsl client

### LINKED ISSUE #6086: Improve postgresql_metrics sink: tests, reconnect, wildcard in filters
Copied from initial PR: https://github.com/timberio/vector/pull/5391#issuecomment-761196457

- [x] add more asserts to tests: https://github.com/timberio/vector/pull/5391#discussion_r558581861 -- Implemented in https://github.com/timberio/vector/pull/6110
- [ ] refactor client reconnect (remove `expect`): https://github.com/timberio/vector/pull/5391#discussion_r558510249 -- Implemented in https://github.com/timberio/vector/pull/6415
- [x] support wildcard in filters: https://github.com/timberio/vector/pull/5391#discussion_r558499855 - Implemented in https://github.com/timberio/vector/pull/6118
- [x] edit CODEOWNERS -- Implemented in https://github.com/timberio/vector/pull/6110
- [x] check version on connect, error if not supported (use `SHOW server_version` / `SHOW server_version_num`) -- Implemented in https://github.com/timberio/vector/pull/6110
- [x] Do not connect on `SourceConfig::build` https://github.com/timberio/vector/pull/6110#issuecomment-762313502 -- Implemented in https://github.com/timberio/vector/pull/6246

### PR DESCRIPTION
**Pull request by fanatid**
Close #6086 

This is the last item from the #6086 list. These changes remove `expect`/`unwrap` for postgresql client / version.

### CODE REVIEW

File: src/sources/postgresql_metrics.rs [hunk 1]
```diff
@@ -182,6 +182,111 @@ impl SourceConfig for PostgresqlMetricsConfig {
...
+            let version = version_row
+                .try_get::<&str, &str>("version")
+                .with_context(|| SelectVersionFailed {
+                    endpoint: config_to_endpoint(&self.config),
+                })?;
+            debug!(message = "Connected to server.", endpoint = %config_to_endpoint(&self.config), server_version = %version);
+        }
+
+        // Get server version and check that we support it
+        let row = client
```
Comment (jszwedko) [inline src/sources/postgresql_metrics.rs:263]: Is there a reason we need to query for the version twice (here and above for logging)? It seems like we could just query once unless I'm missing something. Sorry, I realize it was already like this, but it jumped out to me.

**Review by jszwedko — APPROVED** [review]
This is a nice clean-up, thanks @fanatid !

File: src/sources/postgresql_metrics.rs (hunk 1 above)
Comment (fanatid) [reply src/sources/postgresql_metrics.rs:263]: While these queries return nearly the same, the second one much easier to parse:
```
vector=# select version();
┌──────────────────────────────────────────────────────────────────────────────────────────────────────────────────┐
│                                                     version                                                      │
├──────────────────────────────────────────────────────────────────────────────────────────────────────────────────┤
│ PostgreSQL 13.1 (Debian 13.1-1.pgdg100+1) on x86_64-pc-linux-gnu, compiled by gcc (Debian 8.3.0-6) 8.3.0, 64-bit │
└──────────────────────────────────────────────────────────────────────────────────────────────────────────────────┘
(1 row)

Time: 1.967 ms
vector=# show server_version_num;
┌────────────────────┐
│ server_version_num │
├────────────────────┤
│ 130001             │
└────────────────────┘
(1 row)

Time: 1.147 ms
```
also, we call query `select version()` only if `debug` (or lower) log level enabled.

Comment (jszwedko) [reply src/sources/postgresql_metrics.rs:263]: Yeah, I was just thinking that maybe we could just always execute the query, but only log at the debug level. I do see both are available via the catalogue:

```
postgres=# select * from pg_settings where name in ('server_version_num', 'server_version');
        name        |            setting             | unit |    category    |               short_desc                | extra_desc | context  | vartype | source  | min_val | max_val | enumvals |            boot_val            |           
reset_val            | sourcefile | sourceline | pending_restart 
--------------------+--------------------------------+------+----------------+-----------------------------------------+------------+----------+---------+---------+---------+---------+----------+--------------------------------+-----------
---------------------+------------+------------+-----------------
 server_version     | 13.1 (Debian 13.1-1.pgdg100+1) |      | Preset Options | Shows the server version.               |            | internal | string  | default |         |         |          | 13.1 (Debian 13.1-1.pgdg100+1) | 13.1 (Debi
an 13.1-1.pgdg100+1) |            |            | f
 server_version_num | 130001                         |      | Preset Options | Shows the server version as an integer. |            | internal | integer | default | 130001  | 130001  |          | 130001                         | 130001    
                     |            |            | f
(2 rows)
```

So it seems feasible to do it in one query, but I'm ok with it as-is given that the additional query only happens if debug is enabled.

**Pull request merged**